src/app/_core/services/form-utils.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+} from '@angular/forms';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FormUtilsService {
+  isFieldInError<T extends object>(
+    /**
+     * Détecte si un champ est en erreur dans un FormGroup typé
+     */
+    form: FormGroup<{ [K in keyof T]: FormControl<T[K]> }>,
```

Comment:
Il y a pas moyen de rendre ca plus lisible ?
Ou creer un Objet plus claire à la lecture peut etre ?

Je vois un truc simple pour facilité la lecture : 
En gardant le typage comme tu le fais, mais en le sortant dans un type :
```suggestion
type TFormGroup<T> = FormGroup< [K in keyof T]: FormControl<T[K]> }>;

    form: FormGroup<TFormGroup>,
```